plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -207,24 +208,37 @@ def is_struct_included(struct1, struct2, exclude=None):
             Return True if all element of dict 1 are present in dict 2, return false otherwise.
     """
     if isinstance(struct1, list) and isinstance(struct2, list):
+        if len(struct1) == 0 and len(struct2) == 0:
+            return True
+        found = False
         for item1 in struct1:
-            if isinstance(item1, (list, dict)):
+            found = False
+            if isinstance(item1, list) or isinstance(item1, dict):
+                found1 = False
                 for item2 in struct2:
-                    if not is_struct_included(item1, item2, exclude):
-                        return False
+                    if is_struct_included(item1, item2, exclude):
+                        found1 = True
+                if found1:
+                    found = True
             else:
                 if item1 not in struct2:
                     return False
-        return True
+                else:
+                    found = True
+            if not found:
+                return False
+        return found
```

Comment:
The whole block above - lines 213 to 230 - can be replaced by
```python
        for item1 in struct1:
            if isinstance(item1, (list, dict)):
                for item2 in struct2:
                    if is_struct_included(item1, item2, exclude):
                        break
                else:
                    return False
            else:
                if item1 not in struct2:
                    return False
        return True
```
That's basically identical to the code that was there before, with a small difference:
```diff
diff --git a/plugins/module_utils/identity/keycloak/keycloak.py b/plugins/module_utils/identity/keycloak/keycloak.py
index 20314ede..86d7e49c 100644
--- a/plugins/module_utils/identity/keycloak/keycloak.py
+++ b/plugins/module_utils/identity/keycloak/keycloak.py
@@ -219,8 +219,10 @@ def is_struct_included(struct1, struct2, exclude=None):
         for item1 in struct1:
             if isinstance(item1, (list, dict)):
                 for item2 in struct2:
-                    if not is_struct_included(item1, item2, exclude):
-                        return False
+                    if is_struct_included(item1, item2, exclude):
+                        break
+                else:
+                    return False
             else:
                 if item1 not in struct2:
                     return False
```
This is IMO easier to understand (assuming one knows about https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops).